### PR TITLE
Trace linear as aten::linear

### PR DESCRIPTION
**Pull request by eellison**
https://github.com/pytorch/pytorch/pull/51613 made `torch.nn.functional.linear` compile as `aten::linear`, extend the same behavior with tracing.

### CODE REVIEW

**Comment by eellison** [comment]
cc @BowenBao, i'm running into ONNX failure here in `test_shufflenet_v2_dynamic_axes`: 

> Feb 08 22:52:54 E           onnxruntime.capi.onnxruntime_pybind11_state.RuntimeException: [ONNXRuntimeError] : 6 : RUNTIME_EXCEPTION : Non-zero status code returned while running If node. Name:'If_746' Status Message: Non-zero status code returned while running Gemm node. Name:'Gemm_747' Status Message: /onnxruntime_src/onnxruntime/core/framework/execution_frame.cc:64 onnxruntime::common::Status onnxruntime::IExecutionFrame::GetOrCreateNodeOutputMLValue(int, const onnxruntime::TensorShape*, OrtValue*&, size_t) shape && tensor.Shape() == *shape was false. OrtValue shape verification failed. Current shape:{1,1000} Requested shape:{3,1000}

**Review by Krovatkin — APPROVED** [review]
:shipit:

File: test/onnx/expect/TestOperators.test_linear.expect [hunk 1]
```diff
@@ -1,43 +1,34 @@
...
+    name: "bias"
+    raw_data: "\324BO\276@\245T>\350\377\245\275\374u\336\276&\212\304>"
   }
   initializer {
+    dims: 4
     dims: 5
     data_type: 1
-    name: "bias"
-    raw_data: "\324BO\276@\245T>\350\377\245\275\374u\336\276&\212\304>"
+    name: "6"
```
Comment (BowenBao) [inline test/onnx/expect/TestOperators.test_linear.expect:30]: Is there a way to keep the parameter name for `weight` (instead of `6`), like `bias` above?

Comment (eellison) [reply test/onnx/expect/TestOperators.test_linear.expect:30]: This is the traced graph:

```
graph(%self : __torch__.torch.nn.modules.linear.Linear,
      %input : Float(3, 4, strides=[4, 1], requires_grad=0, device=cpu)):
  %8 : Tensor = prim::GetAttr[name="bias"](%self)
  %7 : Tensor = prim::GetAttr[name="weight"](%self)
  %6 : Float(3, 5, strides=[5, 1], requires_grad=1, device=cpu) = aten::linear(%input, %7, %8) # /private/home/eellison/miniconda3/lib/python3.7/site-packages/torch/nn/functional.py:1753:0
  return (%6)
  ```

I think ONNX could just have a pass that sets the debugName of `prim::GetAttr` nodes to be the attribute name. What do you think ?

Comment (BowenBao) [reply test/onnx/expect/TestOperators.test_linear.expect:30]: after looking at the history for a bit, it looks like the renaming happens due to that constant folding kicked in for weight + transpose, since onnx::matmul is exported instead of onnx::gemm. 
For this i guess a tweak on the test case adding flag `do_constant_folding=False` when calling `_export` should work. 

I'm curious about the previous error with shuffle net. Let me take a look why it didn't work.

Comment (eellison) [reply test/onnx/expect/TestOperators.test_linear.expect:30]: The shufflenet error was before i removed the if statement on the aten::linear lowering

Comment (BowenBao) [reply test/onnx/expect/TestOperators.test_linear.expect:30]: Tried a few steps, that one is a legit issue on how we handle if node dynamic axes export. Could you help open a issue? You can land this one if the above suggestion works, and we can submit a fix afterwards.

Comment (eellison) [reply test/onnx/expect/TestOperators.test_linear.expect:30]: The shufflenet error has been fixed on this PR (there may still be a general issue with dynamic axes). Where do you want me to open issue ?

Comment (BowenBao) [reply test/onnx/expect/TestOperators.test_linear.expect:30]: I was thinking an issue summarizing the error when linear lowers to if statement of addmm and matmul. It's ok we can refer to this PR for tracking as well.

**Review by BowenBao — APPROVED** [review]

File: test/onnx/expect/TestOperators.test_linear.expect (hunk 1 above)
Comment (eellison) [reply test/onnx/expect/TestOperators.test_linear.expect:30]: Issue in onnxruntime or in pytorch i meant ?

Comment (BowenBao) [reply test/onnx/expect/TestOperators.test_linear.expect:30]: pytorch, the issue looks like in exporter, we are working on it.
